### PR TITLE
add softmax, softmax_backward, nll_loss_forward and related op patterns

### PR DESCRIPTION
**Pull request by Yancey1989**
related issue #473

### CODE REVIEW

File: pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp [hunk 1]
```diff
@@ -117,6 +117,7 @@ LogicalResult ConvertAtenOp<AtenNativeDropoutOp>::matchAndRewrite(
   rewriter.replaceOp(op, {output, boolMask});
   return success();
 }
+
```
Comment (fortianyou) [inline pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscDecomposeComplexOps.cpp:120]: Revert this modification?

File: pytorch_blade/tests/disc/ops/test_nn_ops.py [hunk 2]
```diff
@@ -22,11 +23,14 @@ def _test_nn_ops(self, nn_ops_func, x=None):
             test_data = (test_data.to(self.device),)
         self._test_cvt_to_disc(nn_ops_func, test_data)
 
-    @skipTorchGE("1.12.0")
     def test_softmax(self):
         softmax = torch.nn.Softmax(dim=-1)
         self._test_nn_ops(softmax)
 
+        if utils.torch_version_number() >= utils.parse_version("1.12.0"):
```
Comment (fortianyou) [inline pytorch_blade/tests/disc/ops/test_nn_ops.py:30]: Nit： Is this update needed?

Comment (Yancey1989) [reply pytorch_blade/tests/disc/ops/test_nn_ops.py:30]: Yes, i32 input suporting is not in this PR.

**Review by fortianyou — APPROVED** [review]
LGTM
